app/controllers/lessons_controller.rb

```diff
@@ -47,42 +47,46 @@ def new
   end
 
   def create
-    lesson = Lesson.new
-    lesson.hash_tag = params[:lesson][:hash_tag]
-    lesson.created_at = Time.now
-    lesson.category = params[:lesson][:category]
-    lesson.description = params[:lesson][:description]
-    lesson.approved = true
-    lesson.save
-    cookies["new_lesson_id"] = lesson.id
+    if user_signed_in?
+      lesson = Lesson.new
+      lesson.hash_tag = params[:lesson][:hash_tag]
+      lesson.created_at = Time.now
+      lesson.category = params[:lesson][:category]
+      lesson.description = params[:lesson][:description]
+      lesson.approved = true
+      lesson.save
+      cookies["new_lesson_id"] = lesson.id
 
     # create a new contribution for creator
-    contribution = Contribution.new
-    contribution.lesson_id = lesson.id
-    contribution.user_id = session["user_id"]
-    contribution.creator = true
-    contribution.save
+      contribution = Contribution.new
+      contribution.lesson_id = lesson.id
+      contribution.user_id = session["user_id"]
+      contribution.creator = true
+      contribution.save
+    end
+      redirect_to tweets_url(id:lesson.id)
 
-    redirect_to tweets_url(id:lesson.id)
   end
 
   def edit
     @lesson = Lesson.find_by(id: params[:id])
   end
 
   def update
-    @lesson.hash_tag = params[:lesson][:hash_tag]
-    @lesson.created_at = params[:lesson][:date]
-    @lesson.description = params[:lesson][:description]
-    @lesson.approved = true
-    @lesson.save
-
-    # create a new contribution for editor
-    contribution = Contribution.new
-    contribution.lesson_id = @lesson.id
-    contribution.user_id = session["user_id"]
-    contribution.creator = false
-    contribution.save
+    if user_signed_in?
+      @lesson.hash_tag = params[:lesson][:hash_tag]
```

Comment:
I again realize this was not part of your work, but this can also be refactored to use `.update_attributes`, which takes in params (see this [StackOverflow post](http://stackoverflow.com/questions/26090666/rails-update-object-from-params)). Also why is this `created_at` attribute modified? That should not be mutated after creation, otherwise it doesn't serve its purpose.